spring-kafka/src/main/java/org/springframework/kafka/annotation/RetryableTopicAnnotationProcessor.java

```diff
@@ -368,8 +366,8 @@ private List<Class<? extends Throwable>> resolveClasses(Class<? extends Throwabl
 	}
 
 	private Object resolveExpression(String value) {
-		String resolved = resolve(value);
 		if (this.expressionContext != null) {
+			String resolved = resolve(value);
```

Comment:
seems the `resolved` is only used within the if block, so it makes more sense to move it within, so when `expressionContext` is null, no unnecessary overhead will incur.